From 5337703cd7285b2de202d84b3f6139ede6c2dc32 Mon Sep 17 00:00:00 2001
From: Roman Stratiienko <r.stratiienko@gmail.com>
Date: Mon, 26 Dec 2022 10:40:36 +0200
Subject: [PATCH 06/15] drm_hwcomposer: Make compositor frontend-agnostic

Replace 2 remaining structs (hwc_frect_t, hwc_rect_t) with internal
FRect and IRect structures.

Change-Id: I21c097687def7e367eecf89a05c72002dd3e0d7b
Signed-off-by: Roman Stratiienko <r.stratiienko@gmail.com>
---
 compositor/LayerData.h   | 17 ++++++++++++-----
 hwc2_device/HwcLayer.cpp | 14 ++++++++++++--
 2 files changed, 24 insertions(+), 7 deletions(-)

diff --git a/compositor/LayerData.h b/compositor/LayerData.h
index 92d4da045717b1..ff61c8bcb52ae9 100644
--- a/compositor/LayerData.h
+++ b/compositor/LayerData.h
@@ -16,9 +16,6 @@
 
 #pragma once
 
-#include <hardware/hardware.h>
-#include <hardware/hwcomposer.h>
-
 #include <cmath>
 #include <cstdbool>
 #include <cstdint>
@@ -42,11 +39,21 @@ enum LayerTransform : uint32_t {
   kRotate270 = 1 << 4,
 };
 
+struct FRect {
+  float top, bottom;
+  float left, right;
+};
+
+struct IRect {
+  int top, bottom;
+  int left, right;
+};
+
 struct PresentInfo {
   LayerTransform transform{};
   uint16_t alpha = UINT16_MAX;
-  hwc_frect_t source_crop{};
-  hwc_rect_t display_frame{};
+  FRect source_crop{};
+  IRect display_frame{};
 
   bool RequireScalingOrPhasing() const {
     const float src_width = source_crop.right - source_crop.left;
diff --git a/hwc2_device/HwcLayer.cpp b/hwc2_device/HwcLayer.cpp
index 589a0a7144a7be..055ec7236ce07f 100644
--- a/hwc2_device/HwcLayer.cpp
+++ b/hwc2_device/HwcLayer.cpp
@@ -104,7 +104,12 @@ HWC2::Error HwcLayer::SetLayerDataspace(int32_t dataspace) {
 }
 
 HWC2::Error HwcLayer::SetLayerDisplayFrame(hwc_rect_t frame) {
-  layer_data_.pi.display_frame = frame;
+  layer_data_.pi.display_frame = (IRect){
+      .top = frame.top,
+      .bottom = frame.bottom,
+      .left = frame.left,
+      .right = frame.right,
+  };
   return HWC2::Error::None;
 }
 
@@ -121,7 +126,12 @@ HWC2::Error HwcLayer::SetLayerSidebandStream(
 }
 
 HWC2::Error HwcLayer::SetLayerSourceCrop(hwc_frect_t crop) {
-  layer_data_.pi.source_crop = crop;
+  layer_data_.pi.source_crop = (FRect){
+      .top = crop.top,
+      .bottom = crop.bottom,
+      .left = crop.left,
+      .right = crop.right,
+  };
   return HWC2::Error::None;
 }
 
-- 
2.37.2

